Core/include/Acts/Detector/GeometryCompatibilityConcept.hpp

```diff
@@ -14,36 +14,15 @@
 
 namespace Acts::Concepts {
 
-// Types to check compatibility of
-template <typename propagator_state_t, typename navigator_t>
-using ReturnTypeCurrentVolume =
-    decltype(std::declval<navigator_t>().currentVolume(
-        std::declval<propagator_state_t>().navigation));
-
 /// @brief Concept ensuring compatibility TrackingGeometry
 /// and Detector navigation interfaces with the client code
 /// @tparam propagator_state_t Type of the object for navigation state
 /// @tparam navigator_t Type of the navigator object
 template <typename propagator_state_t, typename navigator_t>
-struct NavigationCompatibilityConceptImpl {
-  /// @brief Ensure that the currentVolume method
-  /// returns one of the known volume types
-  constexpr static bool isCurrentVolumePtr =
-      (Acts::Concepts::identical_to<const TrackingVolume*,
-                                    ReturnTypeCurrentVolume, propagator_state_t,
-                                    navigator_t> ||
-       Acts::Concepts::identical_to<const Acts::Experimental::DetectorVolume*,
-                                    ReturnTypeCurrentVolume, propagator_state_t,
-                                    navigator_t>);
-
-  static_assert(isCurrentVolumePtr,
-                "Return type is not a known volume pointer type");
-
-  constexpr static bool value = Acts::Concepts::require<isCurrentVolumePtr>;
+concept NavigationCompatible = requires {
+  requires requires(propagator_state_t & p, navigator_t & n) {
```

Comment:
`requires requires`? 🙀

Reply:
Yeah, unfortunately this is the syntax for putting `requires` blocks inside requires blocks. Concept syntax is a bit bonkers... But now that you say it I think this could be simplified to just:

```cpp
concept NavigationCompatible = requires(propagator_state_t & p, navigator_t & n) {
```

Reply:
Yes; thanks! I was able to clean up the syntax a bit using this reminder.